test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -121,13 +147,12 @@ func (sm *StackManager) DeleteStack(ctx context.Context, name string) error {
 // ForceRemoveStackConfiguration removes the configuration files pulumi creates for managing a stack.
 // It DOES NOT perform any cleanup of the resources created by the stack. Call `DeleteStack` for correct cleanup.
 func (sm *StackManager) ForceRemoveStackConfiguration(ctx context.Context, name string) error {
-	sm.lock.Lock()
-	defer sm.lock.Unlock()
 
-	stack, ok := sm.stacks[name]
+	storedStack, ok := sm.stacks.Get(name)
 	if !ok {
 		return fmt.Errorf("unable to remove stack %s: stack not present", name)
 	}
+	stack := storedStack
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why do you copy the stack reference ?